llbutil/copyop.go

```diff
@@ -26,14 +27,22 @@ func CopyOp(srcState llb.State, srcs []string, destState llb.State, dest string,
 		baseCopyOpts = append(baseCopyOpts, llb.WithCreatedTime(*defaultTs()))
 	}
 	for _, src := range srcs {
+		if ifExists {
+			// If the copy came in as optional (ifExists), then we need to trigger the
+			// underlying wildcard matching and allow empty wildcards. The matching uses
+			// the filepath.Match syntax, so by simply creating a wildcard where the
+			// first letter needs to match the current first letter gets us the single
+			// match; and no error if it is missing.
```

Comment:
Admittedly a hack. But its clean and airtight after much spelunking and experimenting.

Reply:
See here for the matching details: https://golang.org/pkg/path/filepath/#Match